### PR TITLE
Use token to authenticate API request

### PR DESCRIPTION
**Pull request by laritakr**
Fix logic so API token will mimic token's user to grant access to items.

### CODE REVIEW

File: app/models/curate/user_behavior.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ module UserBehavior
     # Format the json for select2 which requires just an id and a field called text.
     # If we need an alternate format we should probably look at a json template gem
     def as_json(opts = nil)
+      require 'byebug'; debugger; true
```
Comment (dbrower) [inline app/models/curate/user_behavior.rb]: Is this line committed on purpose?

File: app/controllers/api/items_controller.rb [hunk 2]
```diff
@@ -51,4 +56,14 @@ def set_current_user!
...
     end
+
+    def validate_permissions!
+      set_current_user!
+      item_id = params[:id]
+      user_name = @current_user.try(:username) || @current_user
+      # if unauthorized
+      respond_with item do |format|
+        format.jsonld { render json: { error: 'Unauthorized', user: user_name, item: item_id }, status: 401 }
+      end unless current_ability.can? :read, item_id
```
Comment (dbrower) [inline app/controllers/api/items_controller.rb]: one nit....the multiline block with a trailing `unless...` clause is confusing. Can the conditional be pulled to the top?

File: app/models/curate/user_behavior.rb (hunk 1 above)
Comment (laritakr) [reply app/models/curate/user_behavior.rb]: fixed and force-pushed

File: app/controllers/api/items_controller.rb (hunk 2 above)
Comment (laritakr) [reply app/controllers/api/items_controller.rb]: changed/simplified this along with added specs

**Pull request merged**